s3_storage_client_test.go

```diff
@@ -285,6 +285,22 @@ func TestGetHeaderUnexpectedError(t *testing.T) {
 	}
 }
 
+func TestGetHeaderNonApiError(t *testing.T) {
+	othErr := errors.New("something went wrong here")
+	meta := fileMetadata{
+		client:    s3.New(s3.Options{}),
+		stageInfo: &execResponseStageInfo{Location: ""},
+		mockHeader: mockHeaderAPI(func(ctx context.Context, params *s3.HeadObjectInput, optFns ...func(*s3.Options)) (*s3.HeadObjectOutput, error) {
+			return nil, othErr
+		}),
+	}
+
+	header, err := new(snowflakeS3Client).getFileHeader(&meta, "file.txt")
+	assertEqualF(t, header, nil, fmt.Sprintf("expected nil header, got: %v", header))
+	assertNotNilF(t, err, "expected err to not be nil")
+	assertEqualF(t, meta.resStatus, errStatus, fmt.Sprintf("expected %v result status for non-APIerror, got: %v", errStatus, meta.resStatus))
```

Comment:
After you rewrote it to use our asserts I have one suggestion - we can use functions ending with `E` (error) instead of `F` (fatal). The difference is that when `E` fails, the test is marked as failed, but it is continuing to check remaining assertions.

Reply:
noted, thank you ! replaced with the Error version